cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/io/cdap/cdap/etl/api/sql/engine/dataset/SparkRecordCollectionImpl.java

```diff
@@ -16,22 +16,27 @@
 
 package io.cdap.cdap.etl.api.sql.engine.dataset;
 
+import io.cdap.cdap.api.spark.sql.DataFrames;
 import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.types.StructType;
+
 
 /**
  * Implementation for SparkRecordCollection
  * @param <T> type of records represented by this collection.
  */
 public class SparkRecordCollectionImpl<T> implements SparkRecordCollection<T> {
 
-  private final JavaRDD<T> javaRDD;
+  private final Dataset<Row> dataset;
```

Comment:
please name it `dataFrame` as `Dataset<Row>` is DataFrame in Spark (often referred as DF): https://spark.apache.org/docs/latest/api/scala/org/apache/spark/sql/index.html#DataFrame=org.apache.spark.sql.Dataset[org.apache.spark.sql.Row]

Reply:
Also it will prevent confusion with CDAP datasets